local-rest-scorer/src/test/java/ai/h2o/mojos/deploy/local/rest/controller/ModelsApiControllerTest.java

```diff
@@ -0,0 +1,174 @@
+  }
+
+  @Test
+  void verifyCapabilities_DefaultShapley_ReturnsExpected() {
+    // Given
+    List<CapabilityType> expectedCapabilities = Arrays.asList(CapabilityType.SCORE);
+
+    MojoScorer scorer = mock(MojoScorer.class);
+    when(scorer.getEnabledShapleyTypes()).thenReturn(ShapleyLoadOption.NONE);
+
+    ModelsApiController controller = new ModelsApiController(scorer, sampleRequestBuilder);
+
+    // When
+    ResponseEntity<List<CapabilityType>> response = controller.getCapabilities();
+
+    // Then
+    assertEquals(expectedCapabilities, response.getBody());
+  }
+
+  @Test
+  void verifyCapabilities_AllShapleyEnabled_ReturnsExpected() {
+    // Given
+    List<CapabilityType> expectedCapabilities = Arrays.asList(
+        CapabilityType.SCORE,
+        CapabilityType.CONTRIBUTION_ORIGINAL,
+        CapabilityType.CONTRIBUTION_TRANSFORMED);
+    MojoScorer scorer = mock(MojoScorer.class);
+    when(scorer.getEnabledShapleyTypes()).thenReturn(ShapleyLoadOption.ALL);
+
+    ModelsApiController controller = new ModelsApiController(scorer, sampleRequestBuilder);
+
+    // When
+    ResponseEntity<List<CapabilityType>> response = controller.getCapabilities();
+
+    // Then
+    assertEquals(expectedCapabilities, response.getBody());
+  }
+
+  @Test
+  void verifyCapabilities_OriginalShapleyEnabled_ReturnsExpected() {
+    // Given
+    List<CapabilityType> expectedCapabilities = Arrays.asList(
+        CapabilityType.SCORE,
+        CapabilityType.CONTRIBUTION_ORIGINAL);
+    MojoScorer scorer = mock(MojoScorer.class);
+    when(scorer.getEnabledShapleyTypes()).thenReturn(ShapleyLoadOption.ORIGINAL);
+
+    ModelsApiController controller = new ModelsApiController(scorer, sampleRequestBuilder);
+
+    // When
+    ResponseEntity<List<CapabilityType>> response = controller.getCapabilities();
+
+    // Then
+    assertEquals(expectedCapabilities, response.getBody());
+  }
+
+  @Test
+  void verifyCapabilities_TransformedShapleyEnabled_ReturnsExpected() {
+    // Given
+    List<CapabilityType> expectedCapabilities = Arrays.asList(
+        CapabilityType.SCORE,
+        CapabilityType.CONTRIBUTION_TRANSFORMED);
+    MojoScorer scorer = mock(MojoScorer.class);
+    when(scorer.getEnabledShapleyTypes()).thenReturn(ShapleyLoadOption.TRANSFORMED);
+
+    ModelsApiController controller = new ModelsApiController(scorer, sampleRequestBuilder);
+
+    // When
+    ResponseEntity<List<CapabilityType>> response = controller.getCapabilities();
+
+    // Then
+    assertEquals(expectedCapabilities, response.getBody());
+  }
+
+  /** Dummy pipeline {@link MojoPipeline} just to mock the static methods used inside scoring. */
+  private static class DummyPipeline extends MojoPipeline {
+    private final MojoFrameMeta inputMeta;
+    private final MojoFrameMeta outputMeta;
+
+    private DummyPipeline(String uuid, MojoFrameMeta inputMeta, MojoFrameMeta outputMeta) {
+      super(uuid, null, null);
+      this.inputMeta = inputMeta;
```

Comment:
Take the following with a grain of salt, since it's usually not a good idea to mock objects you don't own:

Since you're using `DummyPipeline` in this test really just to get around one particular implementation issue (that we auto-load a pipeline as part of a `MojoScorer`), and aren't using this mock for anything really "mocky" ... you might be able to get away with defining a mock `MojoPipeline` (`@Mock private MojoPipeline dummyPipeline`)  rather than define a `DummyPipeline`.

Not sure what other issues might have come up, just sharing general tip.

Reply:
LOVE :heart_eyes: this definitely worked.